diff --git a/os/various/lwip_bindings/arch/cc.h b/os/various/lwip_bindings/arch/cc.h
index eb47666..18c2b36 100644
--- a/os/various/lwip_bindings/arch/cc.h
+++ b/os/various/lwip_bindings/arch/cc.h
@@ -68,7 +68,10 @@ typedef uint32_t        mem_ptr_t;
   osalSysHalt(x);                                                          \
 }
 
+#ifndef BYTE_ORDER
 #define BYTE_ORDER LITTLE_ENDIAN
+#endif
+
 #define LWIP_PROVIDE_ERRNO
 
 #endif /* __CC_H__ */
