diff --git a/os/various/lwip_bindings/lwipthread.c b/os/various/lwip_bindings/lwipthread.c
index 56442ba..b034977 100644
--- a/os/various/lwip_bindings/lwipthread.c
+++ b/os/various/lwip_bindings/lwipthread.c
@@ -279,7 +279,12 @@ static THD_FUNCTION(lwip_thread, p) {
           tcpip_callback_with_block((tcpip_callback_fn) netif_set_link_up,
                                      &thisif, 0);
 #if LWIP_DHCP
-          dhcp_start(&thisif);
+          // If we have no config, run DHCP. Else just do DHCP INFORM.
+          if(ip.addr == 0) {
+            dhcp_start(&thisif);
+          } else {
+            dhcp_inform(&thisif);
+          }
 #endif
         }
         else {
